Code review by "Rautelsi"   (Shashi Rauteloa

snayal opened the pull request:
Attached: code review assignment pdf file.
[Code Review_Rautelsi.pdf](https://github.com/pratibhatiwarii/IS7024-FinalProject/files/6253839/Code.Review_Rautelsi.pdf)

snayal committed:
GoogleAPI text file is missing

snayal committed:
renamed namespace and classes based on model name

snayal committed:
Code Review by Bearcat ID:  rautelsi

discospiff commented on JustBrewIt/BreweryAPI.cs:
```diff
@@ -28,7 +28,7 @@ namespace JustBrewItQuickType
     using Newtonsoft.Json.Converters;
 
-    public partial class Welcome
+    public partial class JustBrewIT
```
This looks a bit more descriptive.

snayal replied:
I have updated the changes.

discospiff commented on JustBrewIt/MyBreweryAPI.cs:
```diff
@@ -77,14 +77,14 @@ public enum Country { UnitedStates };
+    public partial class BreweryAPI
     {
-        public static Welcome[] FromJson(string json) => JsonConvert.DeserializeObject<Welcome[]>(json, QuickType.Converter.Settings);
+        public static BreweryAPI[] FromJson(string json) => JsonConvert.DeserializeObject<BreweryAPI[]>(json, BreweryQuickType.Converter.Settings);
```
Is it a BreweryAPI, or a Brewery?

BreweryAPI == service class that does something.  A verb.
Brewery: a noun, what a service class will return.

This feels more like a noun.

snayal replied:
Thank you prof for the feedback.  I have updated the class and variable.

discospiff commented on JustBrewIt/Pages/Brewery.cshtml.cs:
```diff
@@ -19,7 +19,7 @@ public void OnGet()
             {
                 string jsonString = webClient.DownloadString("https://api.openbrewerydb.org/breweries?by_city=cincinnati&brewery_type=regional");
-                Welcome[] welcome = Welcome.FromJson(jsonString);
+                BreweryAPI[] welcome = BreweryAPI.FromJson(jsonString);
```
Change the variable name from Welcome as well.

snayal replied:
Changed the variable name.

discospiff commented on JustBrewIt/Pages/Index.cshtml.cs:
```diff
@@ -1,7 +1,7 @@
 using Microsoft.AspNetCore.Mvc.RazorPages;
 using Microsoft.Extensions.Logging;
-using QuickType;
+using BreweryQuickType;
```
Good to rename, but I'd drop the QuickType suffix.  Maybe this is what you can name BreweryAPI.

snayal replied:
updated namespace.

discospiff commented on JustBrewIt/Pages/Search.cshtml:
```diff
@@ -1,35 +1,34 @@
 @{
-    var BreweryAPI = (JustBrewItQuickType.Welcome[])ViewData["BreweryAPI"];
+
+    ViewData["Title"] = "Search Page";
```
Good.  Go a bit further, and give it context.  "Brewery Search", for instance, as "Search Page" could be on any website.

snayal replied:
updated.

discospiff commented:
Several good changes worth considering.  See comments in Files Changed section.

snayal committed:
updated changes according to Prof Feedback

snayal commented:
Prof Jones,  I have made changes according to your feedback.  Overall this project is in good shape.  

Thank you for your feedback. 
regards
Shashi

snayal commented on JustBrewIt/Pages/Index.cshtml:
```diff
@@ -52,7 +53,7 @@
                             <h2>We are here to HELP.</h2>
                             <p class="text-light font-weight-bolder">Given the COVID-19 outbreak we urge you to stay tuned for the updated guidelines </p><p class="text-light font-weight-bolder">to make informed decisions regarding your travel.</p>
-                            <p><a class="btn btn-lg btn-primary" href="https://www.brewersassociation.org/brewing-industry-updates/covid-19-resource-center/" role="button">See Guidelines</a></p>
+                            <p><a class="btn btn-lg btn-primary" href="https://www.brewersassociation.org/brewing-industry-updates/covid-19-resource-center/" target="_blank" role="button">See Guidelines</a></p>
```
Fixed:  I have added command, now if you click link (guidelines) on you home page,  link will open in new tab/window.  
 
Benefit:  User can see your Home page as well as the outside link.  
 
     Reference:    https://www.w3schools.com/html/html_links.asp

snayal commented on JustBrewIt/Pages/Search.cshtml:
```diff
@@ -1,35 +1,34 @@
 @model JustBrewIt.Pages.SearchModel
 @{
+
+    ViewData["Title"] = "Search Page";
```
Fixed: Added naming for the pages   ViewData["Title"] = "Search Page";    in Search.cshtml page

snayal commented on JustBrewIt/Pages/Search.cshtml:
```diff
@@ -1,35 +1,34 @@
+        <option value="proprietor">Proprietor</option>
+        <option value="closed">Closed</option>
+    </select> &nbsp;&nbsp;
+    <input type="submit" value="Search"  class="btn btn-sm btn-primary" />
```
**Suggestion 2:**  Search input boxes look better in one line.
**Fixed:**  modified alignment by removing extra <div>  and added & nbsp;&nbsp; space between search box and submit button.

**Suggestion 3:**  Search input boxes look better in one line.
**Action:** modified alignment by removing extra <div>  and added & nbsp;&nbsp; space between the search box and submit button.
 
**Suggestion 3:**  I have changed the class from "form-control" to "featurette-heading". You can also replace it by using another class.
 
<input type="text" class="form-control"   name="BreweryCity" id="BreweryCity" />

**Action**:  <input type="text" class="featurette-heading"   name="BreweryCity" id="BreweryCity" />
 
**Suggestion 4:** Instead of the "search" word it is good to replace it with "select"
 
<label class="control-label" for="BreweryType">Search Brewries by Type</label>
	 
**Action:**  <label class="control-label" for="BreweryType">Select  Brewries Type</label>





https://www.jotform.com/blog/how-to-make-a-web-design-look-good/

snayal commented on JustBrewIt/BreweryAPI.cs:
```diff
@@ -19,7 +19,7 @@
 //    var welcome = Welcome.FromJson(jsonString);
 
-namespace JustBrewItQuickType
+namespace JustBrewIt
```
Suggestion 5:  Model pages have default namespace.   Meaningful naming practice improves code readability.
•	 
Action: I have changed the namespace naming in your models.  
Reference: 
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/namespaces/using-namespaces

snayal commented on JustBrewIt/Pages/ViewDetails.cshtml:
```diff
@@ -1,7 +1,7 @@
 @model JustBrewIt.Pages.ViewDetailsModel
 @{
-    ViewData["Title"] = "Privacy Policy";
+    ViewData["Title"] = "View Details";
```
Changed title of Detail page from "Privacy Policy" to View Details.

snayal commented on JustBrewIt/Pages/Brewery.cshtml.cs:
```diff
@@ -19,8 +19,8 @@ public void OnGet()
                 string jsonString = webClient.DownloadString("https://api.openbrewerydb.org/breweries?by_city=cincinnati&brewery_type=regional");
-                BreweryAPI[] welcome = BreweryAPI.FromJson(jsonString);
-                ViewData["MyBreweryAPI"] = welcome;
+                Brewery[] brewery = Brewery.FromJson(jsonString);
```
changed variable name "welcome" to "brewery"

snayal commented:
**Code Review** 
 
**Summarize the three technical changes you made, and why.**

**Technical Issue 1:**   There was an outside website link on the Home page, when clicked it open the outside website page in own website homepage.  It is a good idea to open outside web links in a new window or tab.

**Fixed:**  I have added a command, now if you click the link (guidelines) on your home page, the link will open in a new tab/window.  
 
Benefit:  User can see your Home page as well as the outside link.  
 
     **Reference:**    https://www.w3schools.com/html/html_links.asp
 
**Technical Issue 2:**  Couldn’t open Detailed page due to missing GoogleAPI.txt file.  string key = System.IO.File.ReadAllText("GoogleAPIKey.txt");

**Action:**  Informed the group member about it.     
 
**Technical Change 2:**  To put "placeholder"  attribute inside  <input  >
 
<input type="text" class="form-control"   name="BreweryCity" id="BreweryCity" />
 
**Fixed:**  <input type="text" class="form-control" placeholder="Enter City name "  name="BreweryCity" id="BreweryCity" />
 
**Technical Change 3:**  Search input boxes look better in one line.
**Fixed:** modified alignment by removing extra <div>  and added & nbsp;&nbsp; space between the search box and submit button.
 
**Technical Change 4:**  I have changed the class from ""form-control" to "featurette-heading". You can also replace it by using another class.
 
**Technical Change5:**
<input type="text" class="form-control"   name="BreweryCity" id="BreweryCity" />

:  <input type="text" class="featurette-heading"   name="BreweryCity" id="BreweryCity" />
 
**Technical Change 6:** Instead of the "search" word it is good to replace it with "select"
 
<label class="control-label" for="BreweryType">Search Brewries by Type</label>
 
Action:  <label class="control-label" for="BreweryType">Select  Brewries Type</label>
7 key principles of designing visually appealing websites | The JotForm Blog

**Technical Change7:**  Model pages have default namespace and class name.   Meaningful naming practice improves code readability.
 
**Fixed:** I have changed the namespace and class naming in your models. 
BreweryAPI.cs : Class welcome to JustBrewIT
MyBreweryAPI.cs: Namespace Quicktype to BreweryQuickType & Class Welcome to BreweryAPI

**Reference:** 
•	https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/namespaces/using-namespaces
•	 
Suggestion 8:  Added naming for the pages   ViewData["Title"] = "Search Page";    in Search.cshtml page

**List three items you learned from the code review**
 
1.	Learnt how to use target the webpage in a new window.    When I visited the website there was an outside link and after clicking outside website opened in the home page and I have to look for the back option to reach previous web page.
2.	I learned that it is good practice to have meaningful naming for model, namespace, class, and methods.  It helps in code readability.   And I also changed our default naming for classes and namespace.
3.	To improve coding and webpage display, visited multiple websites which helped me to learn to code.  I find these 3 websites helped me a lot.  Example:  https://www.w3schools.com ;  
https://getbootstrap.com ;  https://docs.microsoft.com
4.	Learnt that we learn more when we do mistakes because only then we do extra research work to overcome the problem.
Was the project available on GitHub on time?
o	Yes, the project was available on GitHub on time.
Does the project compile?
o	Yes, I can compile the project except the “Detail page” was not working initially due to the missing Googel.API.txt file.   They have added it. 
Does the project have adequate documentation?
o	Yes, the project has enough documentation.
